### ISSUE TITLE
test_access.py errors

### ISSUE THREAD

**Issue by gidden** [iiasa/ixmp#179]
I am currently seeing errors on my local machine:

```
---------------------------------------------------------------------------------------------- Captured stdout call -----------------------------------------------------------------------------------------------
Mock server started with pid 16739
2019-08-01 09:21:01,008  INFO at.ac.iiasa.ixmp.Platform:124 - Welcome to the IX modeling platform!
2019-08-01 09:21:01,009  INFO at.ac.iiasa.ixmp.Platform:125 -  connected to database 'jdbc:hsqldb:file:/tmp/pytest-of-gidden/pytest-4/test_check_single_model_access0/testdb/ixmptest' (user: ixmp)...
Mock server terminated
---------------------------------------------------------------------------------------------- Captured stderr call -----------------------------------------------------------------------------------------------
2019-08-01 09:20:59,028 DEBUG:pretenders.server.server:44: Setting pretender timeout: 120
2019-08-01 09:20:59,028 DEBUG:pretenders.server.server:16: Starting maintainer process
2019-08-01 09:20:59,093 INFO:pretenders.client:100: Requesting None pretender. Port:8000 Timeout:None (None)
2019-08-01 09:20:59,093 DEBUG:pretenders.server.server:44: Setting pretender timeout: 120
Bottle v0.12.17 server starting up (using WSGIRefServer())...
Listening on http://0.0.0.0:8000/
Hit Ctrl-C to quit.

2019-08-01 09:20:59,970 INFO:pretenders.server.apps.pretender:92: Creating http pretender access point at f354b98f-9386-4b30-a7b1-dcc4515d2176 120
127.0.0.1 - - [01/Aug/2019 09:20:59] "POST /http HTTP/1.1" 200 104
127.0.0.1 - - [01/Aug/2019 09:20:59] "POST /preset/f354b98f-9386-4b30-a7b1-dcc4515d2176 HTTP/1.1" 200 0
127.0.0.1 - - [01/Aug/2019 09:20:59] "POST /preset/f354b98f-9386-4b30-a7b1-dcc4515d2176 HTTP/1.1" 200 0
127.0.0.1 - - [01/Aug/2019 09:20:59] "POST /preset/f354b98f-9386-4b30-a7b1-dcc4515d2176 HTTP/1.1" 200 0
SLF4J: Class path contains multiple SLF4J bindings.
SLF4J: Found binding in [jar:file:/home/gidden/.local/lib/python3.7/site-packages/ixmp-0.2.0.post0.dev8-py3.7.egg/ixmp/lib/slf4j-log4j12-1.7.25.jar!/org/slf4j/impl/StaticLoggerBinder.class]
SLF4J: Found binding in [jar:file:/home/gidden/.local/lib/python3.7/site-packages/ixmp-0.2.0.post0.dev8-py3.7.egg/ixmp/lib/slf4j-log4j12-1.7.26.jar!/org/slf4j/impl/StaticLoggerBinder.class]
SLF4J: See http://www.slf4j.org/codes.html#multiple_bindings for an explanation.
SLF4J: Actual binding is of type [org.slf4j.impl.Log4jLoggerFactory]
------------------------------------------------------------------------------------------------ Captured log call ------------------------------------------------------------------------------------------------
INFO     root:core.py:127 launching ixmp.Platform using config file at '/tmp/pytest-of-gidden/pytest-4/test_check_single_model_access0/testdb/test.properties'
__________________________________________________________________________________________ test_check_multi_model_access __________________________________________________________________________________________

tmpdir = local('/tmp/pytest-of-gidden/pytest-4/test_check_multi_model_access0'), test_data_path = PosixPath('/home/gidden/work/iiasa/message/ixmp/tests/data')

    def mocked_test(tmpdir, test_data_path):
        proc = Popen(
            ['python', '-m', 'pretenders.server.server', '--host', '0.0.0.0',
             '--port', '8000'])
        print('Mock server started with pid ' + str(proc.pid))
        sleep(1)
        try:
            # test_fn(*args, **kwargs)
>           test_fn(tmpdir, test_data_path)

tests/test_access.py:25: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _
tests/test_access.py:129: in test_check_multi_model_access
    access = mp.check_access('test_user', ['test_model', 'non_existing_model'])
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _

self = <ixmp.core.Platform object at 0x7f0e4c034630>, user = 'test_user', models = ['test_model', 'non_existing_model'], access = 'view'

    def check_access(self, user, models, access='view'):
        """Check access to specific model
    
        Parameters
        ----------
        user: str
            Registered user name
        models : str or list of str
            Model(s) name
        access : str, optional
            Access type - view or edit
        """
    
        if isinstance(models, str):
            return self._jobj.checkModelAccess(user, access, models)
        else:
            models_list = java.LinkedList()
            for model in models:
                models_list.add(model)
>           access_map = self._jobj.checkModelAccess(user, access, models_list)
E           jpype._jclass.java.lang.NoSuchFieldError: EXCLUDE_EMPTY

../../../../.local/lib/python3.7/site-packages/ixmp-0.2.0.post0.dev8-py3.7.egg/ixmp/core.py:360: java.lang.NoSuchFieldError
```

All required libraries are installed, all other tests pass, and I am on the current master branch.

**Comment by zikolach** [comment]
@gidden can you list files located in your `../../../../.local/lib/python3.7/site-packages/ixmp-0.2.0.post0.dev8-py3.7.egg/ixmp/lib` folder?

**Comment by khaeru** [comment]
I also see exceptions on my machine:
```
------------------------------------------------------------ Captured stdout call ------------------------------------------------------------
Mock server started with pid 9889
Mock server terminated
------------------------------------------------------------ Captured stderr call ------------------------------------------------------------
/usr/bin/python: No module named pretenders.server
```

This is because `python` on Ubuntu is still Python 2.7; `python3` is Python 3.x, which is the only supported Python for ixmp. (When using Anaconda, the PATH environment variable is changed so that `python` points to the Anaconda python version, whether 2.7 or 3.x). The test should use [`sys.executable`](https://docs.python.org/3/library/sys.html#sys.executable).

**Comment by zikolach** [comment]
@khaeru thanks for suggestion!

**Comment by zikolach** [comment]
@gidden can you check if following file exists in your python environment `site-packages/ixmp/lib/jersey-common-2.28.jar`?

**Comment by khaeru** [comment]
Closing this because:
- Partially addressed by #181.
- #182 moves the `checkModelAccess()` code that triggers the original reported error.
- #198 changes `test_access.py` turn the mock server from a decorator into a fixture with teardown.

Any future problems for this test should be filed as a new issue.
